app/src/main/java/net/squanchy/schedule/filterschedule/TracksFilter.kt

```diff
@@ -0,0 +1,29 @@
+package net.squanchy.schedule.filterschedule
+
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import net.squanchy.schedule.domain.view.Track
+import net.squanchy.service.repository.TracksRepository
+
+interface TracksFilter {
+
+    fun updateSelectedTracks(newSelectedTracks: Set<Track>)
+
+    val selectedTracks: Observable<Set<Track>>
+}
+
+private val selectedTracksSubject = BehaviorSubject.create<Set<Track>>()
+
+class InMemoryTracksFilter(private val tracksRepository: TracksRepository) : TracksFilter {
+
+    override fun updateSelectedTracks(newSelectedTracks: Set<Track>) {
+        selectedTracksSubject.onNext(newSelectedTracks)
+    }
+
+    override val selectedTracks: Observable<Set<Track>>
+        get() = selectedTracksSubject.startWith(
+            tracksRepository.tracks()
```

Comment:
If the subject has no values, we prepend the first value we get from the `/tracks` collection on the DB, otherwise nothing.

Reply:
right now we _always_ do a request, even when we don't use it, maybe we can instead do:

```kotlin
selectedTrackSubject.startWith(getTracksFromDbIfEmpty())

fun getTracksFromDbIfEmpty() {
  return if (selectedTracksSubject.hasValue()) { Observable.empty() } else { tracksRepository.tracks().map { it.toSet() } }
}
```

Also I feel it's more explicit and easy to read

Reply:
Great point. Went one step ahead 1d2e259e5e365d641915f1d84e167991937a4793